language-plutus-core/src/Language/PlutusCore/Constant/Apply.hs

```diff
@@ -104,8 +102,8 @@ instance Enum SizeVar where
     fromEnum (SizeVar sizeIndex) = sizeIndex
 
 -- | Same as 'makeBuiltin', but returns a 'ConstAppResult'.
-makeConstAppResult :: TypedBuiltinValue Size a -> Quote ConstAppResult
-makeConstAppResult = fmap (maybe ConstAppFailure ConstAppSuccess) . makeBuiltin
+makeConstAppResult :: TypedBuiltinValue Size a -> ConstAppResult
+makeConstAppResult = maybe ConstAppFailure ConstAppSuccess . makeBuiltin
```

Comment:
We can ignore `Quote` while evaluating things, because abstract machines not only do not require the global uniqueness condition to hold, but also violate this condition in the process of evaluation (though, we could probably change that, but why).